### PR TITLE
(#517) Update the allowed tokens prior to a regex

### LINKED ISSUE #517: syntax error on valid file when: if /regex/ in array
puppet-lint22 webserver.pp  
ERROR: Syntax error (try running `puppet parser validate <file>`) on line 7

<pre>
# monitoring for webserver profile
class profile::icinga::webserver (
){
  $locations = hiera_hash('nginx::locations', false)
  if $locations {
    $locations.each |String $location, Hash $location_info| {
        if /^icinga_service_icon_.*/ in $location_info['tag'] {           # This is the line where the error happens
          $service_icon = regsubst(grep($location_info['tag'], '^icinga_service_icon_'), '^icinga_service_icon_(.*)', 'img/logos/\1')[0]
        }
    }
  }
}
</pre>

I run puppet-lint 2.0.0

This code snippet works fine for me with Puppet 3.8.7 and future parser

### PR DESCRIPTION
**Pull request by rnelson0**
Fixes #517 
I am not aware of other tokens besides `if`/`elsif` that are permitted and not currently allowed. Please let me know if there are.

### CODE REVIEW

File: spec/puppet-lint/lexer_spec.rb [hunk 1]
```diff
@@ -840,6 +840,17 @@
...
+    it 'should properly parse when regex follows an if' do
+      tokens = @lexer.tokenise('if /^icinga_service_icon_.*/ in $location_info { }')
+      expect(tokens[2].type).to eq(:REGEX)
+    end
+
+    it 'should properly parse when a regex follows an elsif' do
+      tokens = @lexer.tokenise('if /a/ in $location_info { } elsif /b/ in $location_info { }')
+      expect(tokens[2].type).to eq(:REGEX)
+      expect(tokens[14].type).to eq(:REGEX)
+    end
```
Comment (rnelson0) [inline spec/puppet-lint/lexer_spec.rb]: I'll add a test for `class { /regex/: }` for a negative control.

Comment (rnelson0) [inline spec/puppet-lint/lexer_spec.rb]: Ironically, this doesn't pass `puppet parser validate` so wouldn't be a great test. I can't come up with a good false positive that does pass validation, ironically. It makes me wonder if this check is even needed anymore.

Comment (binford2k) [inline spec/puppet-lint/lexer_spec.rb]: The more I noodle on it, the more I think you might be right. I don't think I'm ready to yank it, but it would be worth evaluating for 3.0.
